backends/backends.c

```diff
@@ -247,102 +247,187 @@ static void backends_main_cleanup(void *ptr) {
+ */
+void backend_set_prometheus_variables(int *default_port
+        ,int (**backend_response_checker)(BUFFER *)
+        ,int (**backend_request_formatter)(BUFFER *, const char *, RRDHOST *, const char *, RRDSET *, RRDDIM *, time_t, time_t, BACKEND_OPTIONS)
+        ,const char *type) {
+    (void)default_port;
+    (void)type;
+    (void)backend_request_formatter;
 
-        if(BACKEND_OPTIONS_DATA_SOURCE(global_backend_options) == BACKEND_SOURCE_DATA_AS_COLLECTED)
-            *backend_request_formatter = format_dimension_collected_opentsdb_telnet;
-        else
-            *backend_request_formatter = format_dimension_stored_opentsdb_telnet;
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
+    *backend_response_checker = process_prometheus_remote_write_response;
+#endif /* ENABLE_PROMETHEUS_REMOTE_WRITE */
+}
 
-        ret = 0;
-    }
-    else if(!strcmp(type, "opentsdb:http") || !strcmp(type, "opentsdb:https")) {
+/**
+ * Set JSON variables
+ *
+ * Set the variables necessaries to work with this specific backend.
+ *
+ * @param default_port  the default port of the backend
+ * @param backend_response_checker function called to check the result.
+ * @param backend_request_formatter function called to format the msessage to the backend
+ * @param type the backend string selector.
+ */
+void backend_set_json_variables(int *default_port
+        ,int (**backend_response_checker)(BUFFER *)
+        ,int (**backend_request_formatter)(BUFFER *, const char *, RRDHOST *, const char *, RRDSET *, RRDDIM *, time_t, time_t, BACKEND_OPTIONS)
+        ,const char *type) {
+    (void)type;
+    *default_port = 5448;
+    *backend_response_checker = process_json_response;
+
+    if (BACKEND_OPTIONS_DATA_SOURCE(global_backend_options) == BACKEND_SOURCE_DATA_AS_COLLECTED)
+        *backend_request_formatter = format_dimension_collected_json_plaintext;
+    else
+        *backend_request_formatter = format_dimension_stored_json_plaintext;
+}
 
-        *default_port = 4242;
-        *backend_response_checker = process_opentsdb_response;
+/**
+ * Set OpenTSDB HTTP variables
+ *
+ * Set the variables necessaries to work with this specific backend.
+ *
+ * @param default_port  the default port of the backend
+ * @param backend_response_checker function called to check the result.
+ * @param backend_request_formatter function called to format the msessage to the backend
+ * @param type the backend string selector.
+ */
+void backend_set_opentsdb_http_variables(int *default_port
+        ,int (**backend_response_checker)(BUFFER *)
+        ,int (**backend_request_formatter)(BUFFER *, const char *, RRDHOST *, const char *, RRDSET *, RRDDIM *, time_t, time_t, BACKEND_OPTIONS)
+        ,const char *type) {
+    *default_port = 4242;
+    *backend_response_checker = process_opentsdb_response;
 
 #ifdef ENABLE_HTTPS
-        if ( *(type+13) == 's') {
-            security_start_ssl(2);
-        }
+    if ( *(type+13) == 's') {
+        security_start_ssl(2);
+    }
```

Comment:
i think `const char *type is not needed in all `backend_set_`, i see you need it only (so far) in `backend_set_opentsdb_telnet_variables.

How about to add `BACKEND_TYPE_OPENTSDB_HTTPS` type? it will solv the problem

Reply:
I think it is not necessary, when you did this question you gave me an idea, give me few minutes and I will upload a new version!

Reply:
ok, but please don;t use `char *type` in set funcs, it is wrong, only `backend_select` should use it

Reply:
I am not sure if it is clear what does const mean when declared on input parameters of a method.  It does not imply visibility, it rather implies value can't be changed.
Check [here](https://en.wikipedia.org/wiki/Const_(computer_programming)#Parameters_and_variables) and you 'll see:
```A const automatic variable (non-static local variable) means that single assignment is happening, though a different value may be used each time, such as const int x_squared = x*x. A const parameter in pass-by-reference means that the referenced value is not modified – it is part of the contract – while a const parameter in pass-by-value (or the pointer itself, in pass-by-reference) does not add anything to the interface (as the value has been copied), but indicates that internally, the function does not modify the parameter (it is a single assignment). For this reason, some favor using const in parameters only for pass-by-reference, where it changes the contract, but not for pass-by-value, where it exposes the implementation.```

Reply:
At this specific line of code, i would only indicate the magic numbers that don't help code readability.